sw/device/mask_rom/sys_state.c

```diff
@@ -0,0 +1,108 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/mask_rom/sys_state.h"
+
+#include <stddef.h>
+
+// The below function represent functions that would come from external
+// dependencies.
+extern bool rom_ext_manifest_get_security_descriptor(rom_ext_manifest_tag_t *);
+
+// Returns true if the buffers are equal.
+bool secure_memcmp(void *a, void *b, size_t len) {
+  char *a8 = (char *)a;
+  char *b8 = (char *)b;
+  char xors = 0;
+  for (int i = 0; i < len; ++i) {
+    xors |= a8[i] ^ b8[i];
+  }
+  return xors == 0;
+}
+
+// Returns true if the buffer is zeroed.
+bool secure_is_zeroed(void *a, size_t len) {
+  char *a8 = (char *)a;
+  char xors = 0;
+  for (int i = 0; i < len; ++i) {
+    xors |= a8[i];
+  }
+  return xors == 0;
+}
+// End external functions.
+
+// Open Q: I think we want this to *not* be static, so that tests can reach
+// into this symbol and emulate glitching.
+//
+// For now, I'm giving it a trailing underscore and a "uniqueish" name, to match
+// how we might write a private member of a class in C++, since that's kind of
+// what this is.
+static rom_ext_manifest_tag_t sys_state_rom_ext_security_descriptor_;
```

Comment:
I wonder if having a DIF-like interface would make our lives easier, i.e. a pointer to a `struct` that holds these values is passed to `init`, which initializes all the values, and other functions compare these values to fresh measurements. WDYT?

Reply:
Maybe. I feel like in some manner it should be the System State module that "owns" these values though. @gkelly @lenary thoughts?